server/resp/src/main/java/org/infinispan/server/resp/commands/generic/TOUCH.java

```diff
@@ -0,0 +1,40 @@
+package org.infinispan.server.resp.commands.generic;
+
+import io.netty.channel.ChannelHandlerContext;
+import org.infinispan.commons.util.concurrent.CompletableFutures;
+import org.infinispan.server.resp.Consumers;
+import org.infinispan.server.resp.Resp3Handler;
+import org.infinispan.server.resp.RespCommand;
+import org.infinispan.server.resp.RespRequestHandler;
+import org.infinispan.server.resp.commands.Resp3Command;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+
+/**
+ * Alters the last access time of a key(s). A key is ignored if it does not exist.
+ * Integer reply: The number of keys that were touched.
+ *
+ * @link https://redis.io/commands/touch/
+ * @since 15.0
+ */
+public class TOUCH extends RespCommand implements Resp3Command {
+   public TOUCH() {
+      super(-2, 1, -1, 1);
+   }
+   @Override
+   public CompletionStage<RespRequestHandler> perform(Resp3Handler handler,
+                                                      ChannelHandlerContext ctx,
+                                                      List<byte[]> arguments) {
+
+      CompletableFuture<Long> totalTouchCount = CompletableFutures.sequence(
```

Comment:
Maybe we should use `CompletionStages.performSequentially` instead? I wonder if we want to have N concurrent touches generated or not. I am open either way.

Reply:
I don't have any strong opinion on this

Reply:
@jabolina ?

Reply:
It might be too much going sequential to mitigate the cases for long lists? We could do something based on the number of keys in the future if needed.

Reply:
I would rather go sequential to begin and add in concurrency limited touches than have an unbounded concurrent amount of touches personally.